### PR TITLE
Promote StartupProbe to beta (enabled by default).

### PR DESCRIPTION
**Pull request by matthyx**
Promote StartupProbe to beta (enabled by default), related to KEP kubernetes/enhancements#950

### CODE REVIEW

**Comment by bradtopol** [comment]
The updates match the discussion in the Kep https://github.com/kubernetes/enhancements/issues/950 and even appear to be on the proper branch! Great job @matthyx!!!
/lgtm
/approve

**Comment by daminisatya** [comment]
Thank you @bradtopol for reviewing

**Comment by matthyx** [comment]
@bradtopol @daminisatya after discussing with @mrbobbytables and @guineveresaenger we have agreed on postponing the beta to 1.18
Can you revert this change, or give guidance on how to proceed?

File: static/docs/reference/generated/kubernetes-api/v1.16/index.html [hunk 1]
```diff
@@ -3886,7 +3886,7 @@ <H1 id="container-v1-core">Container v1 core</H1>
...
+<TR><TD><CODE>startupProbe</CODE><BR /><I><a href="#probe-v1-core">Probe</a></I></TD><TD>StartupProbe indicates that the Pod has successfully initialized. If specified, no other probes are executed until this completes successfully. If this probe fails, the Pod will be restarted, just as if the livenessProbe failed. This can be used to provide different probe parameters at the beginning of a Pod&#39;s lifecycle, when it might take a long time to load data or warm a cache, than during steady-state operation. This cannot be updated. This is an beta feature enabled by default that can be disabled using the StartupProbe feature flag. More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle#container-probes</TD></TR>
```
Comment (sftim) [inline static/docs/reference/generated/kubernetes-api/v1.16/index.html]: This is a change to the v1.16 API documentation. Is that right?

Comment (matthyx) [reply static/docs/reference/generated/kubernetes-api/v1.16/index.html]: This was a change introduced as alpha in 1.16
I was going to promote it to beta in 1.17 until we decided to back off

Comment (sftim) [reply static/docs/reference/generated/kubernetes-api/v1.16/index.html]: I think that will get overwritten the next time someone generates the v1.16 reference documentation

Comment (matthyx) [reply static/docs/reference/generated/kubernetes-api/v1.16/index.html]: good, so we just wait and the issue will resolve by itself

**Pull request merged**